blog/_posts/2024-06-06-istio-native-sidecar.md

```diff
@@ -0,0 +1,167 @@
+```
+
+### Experiment lifecycle management
+
+One benefit of native sidecar support is that Kubernetes properly manages the sidecar lifecycle. Let's conduct an experiment to demonstrate this.
+
+1. Restart application deployment
+
+   ```bash
+   kubectl rollout restart deployment httpbin
+   ```
+
+2. Check pod events and istio-proxy logs
+
+   ```bash
+   kubectl get events --sort-by='.lastTimestamp'
+   ```
+
+   | LAST SEEN | TYPE | REASON | OBJECT | MESSAGE |
+   |-----------|------|--------|--------|---------|
+   | 2m39s | Normal | SuccessfulCreate | replicaset/httpbin-9c5fdf746 | Created pod: httpbin-9c5fdf746-hzmsf |
+   | 2m39s | Normal | ScalingReplicaSet | deployment/httpbin | Scaled up replica set httpbin-9c5fdf746 to 1 |
+   | 2m38s | Normal | Created | pod/httpbin-9c5fdf746-hzmsf | Created container istio-init |
+   | 2m38s | Normal | Created | pod/httpbin-9c5fdf746-hzmsf | Created container istio-proxy |
+   | 2m38s | Normal | Started | pod/httpbin-9c5fdf746-hzmsf | Started container istio-proxy |
+   | 2m38s | Normal | Started | pod/httpbin-9c5fdf746-hzmsf | Started container istio-init |
+   | 2m28s | Normal | Created | pod/httpbin-9c5fdf746-hzmsf | Created container httpbin | 
+   | 2m28s | Normal | Started | pod/httpbin-9c5fdf746-hzmsf | Started container httpbin |
+   | 2m27s | Normal | Killing | pod/httpbin-598f48c6d-nwx67 | Stopping container istio-proxy |
+   | 2m27s | Normal | SuccessfulDelete | replicaset/httpbin-598f48c6d | Deleted pod: httpbin-598f48c6d-nwx67 |
+   | 2m27s | Normal | Killing | pod/httpbin-598f48c6d-nwx67 | Stopping container httpbin |
+
+   We can see, when the pod starts, istio-proxy as an init container starts before the application container. It gives enough time for istio-proxy to warm up before serving traffic.
+
+   istio-proxy used 9 seconds to get ready.
+
+   ```log
+   2024-06-04T05:07:02.138613Z    info    cache    returned workload trust anchor from cache    ttl=23h59m59.861388298s
+   2024-06-04T05:07:10.904779Z    info    Readiness succeeded in 9.015247842s
+   2024-06-04T05:07:10.905752Z    info    Envoy proxy is ready
+   ```
+
+   Events show application container was created 10 seconds after istio-proxy container was created. That is 1 second after the istio-proxy sidecar became ready.
+
+   | LAST SEEN | TYPE | REASON | OBJECT | MESSAGE |
+   |-----------|------|--------|--------|---------|
+   | 2m38s | Normal | Started | pod/httpbin-9c5fdf746-hzmsf | Started container istio-proxy |
+   | 2m38s | Normal | Started | pod/httpbin-9c5fdf746-hzmsf | Started container istio-init |
+   | 2m28s | Normal | Created | pod/httpbin-9c5fdf746-hzmsf | Created container httpbin |
+   | 2m28s | Normal | Started | pod/httpbin-9c5fdf746-hzmsf | Started container httpbin |
+
+   At the termination time, sidecar get killed before the application container.  This gives sidecar time to drain inflight traffic before terminating the application container.
+
+   ```log
```

Comment:
`text` renders better than `log`. Also, is there a command the reader should run to see these logs?

```suggestion
   ```text
```